runtime/j9vm/javanextvmi.cpp

```diff
@@ -464,7 +464,15 @@ JVM_GetClassFileVersion(JNIEnv *env, jclass cls)
 JNIEXPORT void JNICALL
 JVM_VirtualThreadHideFrames(JNIEnv *env, jobject vthread, jboolean hide)
 {
-	/* TODO toggle hiding of stack frames for JVMTI */
+	J9VMThread *currentThread = (J9VMThread *)env;
+	j9object_t threadObject = J9_JNI_UNWRAP_REFERENCE(vthread);
+	j9object_t carrierThread = J9VMJAVALANGVIRTUALTHREAD_CARRIERTHREAD(currentThread, threadObject);
+	J9VMThread *targetThread = J9VMJAVALANGTHREAD_THREADREF(currentThread, carrierThread);
+	if (hide) {
+		targetThread->privateFlags |= J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES;
+	} else {
+		targetThread->privateFlags &= ~(UDATA)J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES;
```

Comment:
```suggestion
	J9VMThread *currentThread = (J9VMThread *)env;
	Assert_SC_true(IS_JAVA_LANG_VIRTUALTHREAD(currentThread, currentThread->threadObject));
	
	if (hide) {
		Assert_SC_true(J9_ARE_NO_BITS_SET(currentThread->privateFlags, J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES);
		currentThread->privateFlags |= J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES;
	} else {
		Assert_SC_true(J9_ARE_ALL_BITS_SET(currentThread->privateFlags, J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES);
		currentThread->privateFlags &= ~(UDATA)J9_PRIVATE_FLAGS_VIRTUAL_THREAD_HIDDEN_FRAMES;
	}
```

Per discussion with @thallium, the segfault seen in testing is due to this API being called on the incorrect receiver object. 

This issue is from JCL code https://github.com/ibmruntimes/openj9-openjdk-jdk20/blob/openj9/src/java.base/share/classes/java/lang/VirtualThread.java#L699-L703 when a virtual thread tries to unpark another virtual thread. The problem is that while the `switchToCarrierThread` is correctly called using the currentThread object, the next call to `switchToVirtualThread` is using the implicit `this` receiver which isn't the currentThread object.

To fix this issue, we can either fix the JCL to use to proper caller (either in `unpark` or `switchToVirtualThread`, or as suggested above to ignore the vthread object passed to native API and always assume that the call should be updating `currentThread` (which is what this API should function and how the current JCL code have implied).

@gacholio do you see any issue with purposed change or should take the JCL patch approach?

Reply:
Ignoring the parameter is likely the better approach - it guarantees that we're operating on the current thread, which means the privateFlags can be used without issue.